### PR TITLE
Update python generate_addresses to be more Pythonic

### LINKED ISSUE #47: Create pythonic generate_addresses function
## Description

Instead of using a dict full of options the args should be passed in a pythonic way.

Part of iotaledger/iota-sdk#58 

## Are you planning to do it yourself in a pull request?

Yes.

### PR DESCRIPTION
**Pull request by Dr-Electron**
# Description of change

This is just a cherry-pick of https://github.com/iotaledger/iota.rs/pull/1651

## Links to any relevant issues

Fixes https://github.com/iotaledger/iota-sdk/issues/47

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

The generated json was validated with the jsonschema generated by my tool

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: sdk/src/client/bindings/python/examples/stronghold.py [hunk 1]
```diff
@@ -11,12 +11,6 @@
...
-address = client.generate_addresses(secret_manager, {
-    "account_index": 0,
-    "range": {
-        "start": 0,
-        "end": 1,
-    },
-})
+address = client.generate_addresses(secret_manager, account_index=0, start=0, end=1)
 
 print(f'Address: {address[0]}')
```
Comment (Dr-Electron) [inline sdk/src/client/bindings/python/examples/stronghold.py]:
```suggestion
print(f'Address: {address[0]}')

```

File: sdk/src/client/bindings/python/examples/ledger_nano.py [hunk 2]
```diff
@@ -17,12 +17,6 @@
...
-address = client.generate_addresses(secret_manager, {
-    "accountIndex": 0,
-    "range": {
-        "start": 0,
-        "end": 1,
-    },
-})
+address = client.generate_addresses(secret_manager, account_index=0, start=0, end=1)
 
 print(f'Address: {address[0]}')
```
Comment (Dr-Electron) [inline sdk/src/client/bindings/python/examples/ledger_nano.py]:
```suggestion
print(f'Address: {address[0]}')

```

File: sdk/src/client/bindings/python/examples/02_generate_addresses.py [hunk 3]
```diff
@@ -8,49 +8,40 @@
...
-addresses = client.generate_addresses(secret_manager, {
-    "coinType": 4219, # Shimmer coin type
-    "accountIndex": 0,
-    "range": {
-        "start": 0,
-        "end": 4,
-    },
-    "internal": False,
+addresses = client.generate_addresses(secret_manager, 
+    coin_type=4219, # Shimmer coin type
```
Comment (Thoralf-M) [inline sdk/src/client/bindings/python/examples/02_generate_addresses.py]: Should we add constants for IOTA and Shimmer coin type to the Python binding?

Comment (Dr-Electron) [reply sdk/src/client/bindings/python/examples/02_generate_addresses.py]: Yeah, I thought about it, but seems like I forgot it 😅 .
Will update this PR

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**